Packages/Core/Editor/Drawers/AtomBaseReferenceDrawer.cs

```diff
@@ -100,5 +103,63 @@ public override void OnGUI(Rect position, SerializedProperty property, GUIConten
             EditorGUI.indentLevel = indent;
             EditorGUI.EndProperty();
         }
+
+        #region Auto DragAndDrop UsageTypeDetection
+        private void DetermineDragAndDropFieldReferenceType(in Rect position, SerializedProperty property,
+            ref int newUsageValue)
+        {
+            EventType mouseEventType = Event.current.type;
+
+            if (mouseEventType != EventType.DragUpdated && mouseEventType != EventType.DragPerform)
+                return;
+
+            if (!IsMouseHoveringOverProperty(position))
+                return;
+
+            var draggedObjects = DragAndDrop.objectReferences;
+
+            if (draggedObjects.Length < 1)
+                return;
+
+            Object draggedObject = draggedObjects[0];
+
+            if (draggedObject is GameObject gameObject)
+                draggedObject = gameObject.GetComponent<TInstancerType>();
```

Comment:
> This drops the amount of script file modifications (from 10 to 4) and make the solution more compact in my point of view. Thanks!

but ... all the other classes also need to be able to be casted to the Interfaces, why would it reduce the amount of file modifications? or am I just missing something?

hm... maybe I should invest a bit time to test it through with all of the 8 different reference drawer types and there variants.

Reply:
Because there are only two types of instancers: the `AtomBaseVariableInstancer` and the `AtomEventInstancer`, all of the instancers inherit either one of these 2, so if we have a common interface for both of them then I don't see what's the problem? The `GetType` always gets the last child class type, and I can be sure of it because my solution (I updated my PR) works. So basically, the drawers don't need to specify any generic class to the `AtomBaseReferenceDrawer` if we work this way.